prov/efa/src/efa_hmem.c

```diff
@@ -142,6 +134,23 @@ static int efa_domain_hmem_info_init_cuda(struct efa_domain *efa_domain)
 	else
 		info->p2p_required_by_impl = true;
 
+	if (ofi_hmem_p2p_disabled()) {
+		info->p2p_disabled_by_user = true;
+		info->p2p_supported_by_device = false;
+		efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_CUDA);
+		return 0;
+	}
+
+	cuda_ret = ofi_cudaMalloc(&ptr, len);
+	if (cuda_ret != cudaSuccess) {
+		info->initialized = false;
+		info->p2p_required_by_impl = true;
```

Comment:
This also doesn't make sense. Failed allocation of CUDA memory doesn't imply any kind of P2P restriction. This feels like overly defensive coding. The P2P requirement for each iface is more of a fixed attribute.